HPC/WRF.md

```diff
@@ -0,0 +1,278 @@
+## WRF
+
+The Weather Research and Forecasting (WRF) model is one of the most used numerical weather prediction (NWP) system. WRF is used extensively for research and real-time forecasting. Large amount of computation resources are required for each simulation, especially for high resolution simulations. Compute and MPI performance are very important for WRF performance.
+
+## WRF correctness fix
+We have observed small numeric differences between WRF output models (v4.2.2) generated on Arm64 and intel platforms. The issue is caused by rounding error at last position (1ulp) from transcendental functions and several compiler optimization options. After identifying the root cause, we implemented [a patch](https://github.com/wrf-model/WRF/pull/1773) that allows bit-by-bit matching results on Graviton and intel processors. This patch works for both v4.2.2 and v4.5 (latest version). 
+This work confirms that the numerical divergence between arm64 and x86_64 is caused by round-up error and unsafe optimizations (FMA, ...) from the compiler. The output models on the two platforms are nearly identical. This type of numerical divergence are likely to show up again when using a different math library, a different version of math library, compiler. 
+If you wish to repeat this experiment, we recommend to use two Ubuntu 18.04 Parallel Clusters, one with arm64 instances (c6gn, c7gn or Hpc7g), and the other with x86_64 instances (c5n or c6i). Follow the instructions below to build WRF.
+
+## Build and run WRF with GCC on arm64
```

Comment:
Is there anything unique about these steps when compared to running WRF on x86? If not, we can delete it.
Just highlight what is the delta if they need to apply for getting best perf for WRF on arm64. For example, number of threads x MPI ranks.

Reply:
Yes that is good. Thanks